### ISSUE TITLE
Cartopy quiver plot issues - custom angles and updating quiver data

### ISSUE THREAD

**Issue by bproxauf** [SciTools/cartopy#1684]
### Description

I experience issues with the `angles` argument and the `set_UVC()` method for quiver plots when using the orthographic projection in Cartopy, see

see https://stackoverflow.com/questions/65021028/cartopy-quiver-plot-issues-custom-angles-and-updating-quiver-data

#### Code to reproduce

```

```

#### Traceback 

```

```

<details>
 <summary>Full environment definition</summary>

### conda list

```
```

### pip list

```
```

</details>

**Comment by robbiemallett** [comment]
Yes, I am also having these issues with quiver in NorthPolarStereo.

It seems that I can only get it to work by calling set_UVC with the data (rather than putting the data in the ax.quiver call.

See: https://stackoverflow.com/questions/66291698/matplotlib-quiver-plot-behaves-differently-after-using-the-set-uvc-method
